src/libs/actions/IOU.js

```diff
@@ -3698,6 +3690,30 @@ function getIOUReportID(iou, route) {
     return lodashGet(route, 'params.reportID') || lodashGet(iou, 'participants.0.reportID', '');
 }
 
+/**
+ * @param {Object} transaction
+ * @param {String} comment
+ * @param {Boolean} isDraft
+ * @param {String} iouType
+ * @param {String} reportID
+ */
+function setMoneyRequestDescription_temporaryForRefactor(transaction, comment, isDraft, iouType, reportID) {
+    // Only update comment if it has changed
+    if (comment.trim() === lodashGet(transaction, 'comment.comment', '')) {
+        return;
+    }
+    if (isDraft) {
+        Onyx.merge(`${ONYXKEYS.COLLECTION.TRANSACTION_DRAFT}${transaction.transactionID}`, {comment: {comment}});
+    } else {
+        if (iouType === CONST.IOU.TYPE.REQUEST) {
+            updateMoneyRequestDescription(transaction.transactionID, reportID, comment);
+        }
+        if (iouType === CONST.IOU.TYPE.SPLIT) {
+            setDraftSplitTransaction(transaction.transactionID, {comment});
+        }
```

Comment:
@s77rt `isDraft` is to determine whether creating or editing request. But let's see the editing split flow

https://github.com/Expensify/App/assets/141406735/f551d551-dbef-4adb-8543-6e5eb70dd909



When saving new description, we will save new comment to draft split transaction, we only call API to update only when user click "Split"

Reply:
We have a similar flow on the waypoint selection and we update the transaction itself


https://github.com/Expensify/App/assets/16493223/837a9b47-9ffc-44ed-90d0-84258d7f5e58

Reply:
@s77rt I do that as we did in the old component here
 
https://github.com/Expensify/App/blob/c5ca9a810b529cdc8b8fceb179e9cfc01061355a/src/pages/EditSplitBillPage.js#L83-L86

This logic is implemented in [this PR](https://github.com/Expensify/App/pull/29064)

Reply:
If we want to change the logic I think we need to confirm with the author of [this PR](https://github.com/Expensify/App/pull/29064)

Reply:
The split bill edit seems to follow a different approach, we edit the draft and use the draft values in completeSplitBill to make the changes. The least we can do here is to call `setMoneyRequestDescription` with `isDraft = true`. Otherwise it will be much confusing to see isDraft false yet still update the drafts collections